quicklogic/qlf_k4n8/utils/repacker/repack.py

```diff
@@ -929,6 +930,48 @@ def load_repacking_constraints(json_root):
 # =============================================================================
 
 
+def load_pcf_constraints(pcf):
+    """
+    Loads constraints for the repacker from a parsed PCF file
+    """
+
+    logging.debug(" Repacking constraints:")
+
+    constraints = []
+    for pcf_constr in parse_simple_pcf(pcf):
+        if (type(pcf_constr).__name__ == 'PcfClkConstraint'):
+
+            # There are only "clb" and "io" tile types
+            # We select the same global clock for
+            # each tile where net is used
+            constraint = RepackingConstraint(
+                net=pcf_constr.net, block_type='clb', port_spec=pcf_constr.pin
+            )
+            constraints.append(constraint)
+
+            logging.debug(
+                "  {}: {}.{}[{}]".format(
+                    constraint.net, constraint.block_type, constraint.port,
+                    constraint.pin
+                )
+            )
+
+            constraint.block_type = 'io'
```

Comment:
I think you have to create a new `RepackingConstraint` object here. Right now this code changes the block type of the existing constraints and adds it again to the list so in the end there are two identical.

Reply:
Fixed